tests/integration/targets/latest/tasks/cloud_resource_type.yml

```diff
@@ -0,0 +1,157 @@
+
+    - name: "1 - Create cloud resource type within Nautobot with only required information"
+      networktocode.nautobot.cloud_resource_type:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        name: Cisco Quantum Network
+        cloud_provider: Cisco
+        content_types:
+          - "cloud.cloudnetwork"
+      register: test_create_min
+
+    - name: "1 - ASSERT"
+      assert:
+        that: 
+          - test_create_min is changed
+          - test_create_min['diff']['before']['state'] == "absent"
+          - test_create_min['diff']['after']['state'] == "present"
+          - test_create_min['cloud_resource_type']['name'] == "Cisco Quantum Network"
+          - test_create_min['msg'] == "cloud_resource_type Cisco Quantum Network created"
+
+    - name: "2 - Duplicate"
+      networktocode.nautobot.cloud_resource_type:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        name: Cisco Quantum Network
+
+      register: test_create_idem
+    - name: "2 - ASSERT"
+      assert:
+        that:
+          - not test_create_idem['changed']
+          - test_create_idem['msg'] == "cloud_resource_type Cisco Quantum Network already exists"
+          - test_create_idem['cloud_resource_type']['name'] == "Cisco Quantum Network"
+    
+    - name: "3 - Update cloud_resource_type"
+      networktocode.nautobot.cloud_resource_type:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        name: Cisco Quantum Network
+        description: Very fast network!
+      register: test_update
+    
+    - name: "3 - ASSERT"
+      assert:
+        that:
+          - test_update is changed
+          - test_update['diff']['before']['description'] == ""
+          - test_update['diff']['after']['description'] == "Very fast network!"
+
+    - name: "4 - Update idempotent"
+      networktocode.nautobot.cloud_resource_type:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        name: Cisco Quantum Network
+        description: Very fast network!
+      register: test_update_idem
+
+    - name: "4 - ASSERT"
+      assert:
+        that:
+          - not test_update_idem['changed']
+          - test_update_idem['msg'] == "cloud_resource_type Cisco Quantum Network already exists"
+          - test_update_idem['cloud_resource_type']['name'] == "Cisco Quantum Network"
+
+    - name: "5 - Create cloud_resource_type with all parameters"
+      networktocode.nautobot.cloud_resource_type:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        name: Cisco Quantum Network Remix
+        cloud_provider: Cisco
+        description: Very fast network reimagined!
+        content_types:
+          - "cloud.cloudnetwork"
+        config_schema:
+          my_config:
+            foo: bar
```

Comment:
You didn't include this in the idempotent test below

Reply:
@joewesch  I think this might be unsupported at the moment.  I consider this very similar to `local_config_context_data` or `napalm_args`.  I cannot find any test cases where an idempotent test is being ran against a `type="dict"` object.   I see some special code to handle `custom_fields` which is similar, so that might be needed here as well.  

Below is the output when attempting it, notice the `before` doesn't pick it up.  I sort of seen that when testing, so I added a 5.1 tasks to query for that object instead. 

```
changed: [localhost] => {
    "changed": true,
    "cloud_resource_type": {
        "config_schema": {
            "my_config": {
                "foo": "bar"
            }
        },
        "content_types": [
            "cloud.cloudnetwork"
        ],
        "created": "2024-11-06T20:25:42.632561Z",
        "custom_fields": {},
        "description": "Very fast network reimagined!",
        "display": "Cisco: Cisco Quantum Network Remix",
        "id": "0238d2a2-72f5-4f63-9e5a-a377382e5ad9",
        "last_updated": "2024-11-06T20:25:42.632576Z",
        "name": "Cisco Quantum Network Remix",
        "natural_slug": "cisco-quantum-network-remix_0238",
        "notes_url": "https://demo.nautobot.com/api/cloud/cloud-resource-types/0238d2a2-72f5-4f63-9e5a-a377382e5ad9/notes/",
        "object_type": "cloud.cloudresourcetype",
        "provider": "b329ec0d-5ee7-47d8-b4a7-3951c28e2979",
        "tags": [],
        "url": "https://demo.nautobot.com/api/cloud/cloud-resource-types/0238d2a2-72f5-4f63-9e5a-a377382e5ad9/"
    },
    "diff": {
        "after": {
            "config_schema": {
                "my_config": {
                    "foo": "bar"
                }
            }
        },
        "before": {
            "config_schema": ""
        }
    },
    "invocation": {
        "module_args": {
            "api_version": null,
            "cloud_provider": "Cisco",
            "config_schema": {
                "my_config": {
                    "foo": "bar"
                }
            },
            "content_types": [
                "cloud.cloudnetwork"
            ],
            "custom_fields": null,
            "description": "Very fast network reimagined!",
            "name": "Cisco Quantum Network Remix",
            "provider": "Cisco",
            "query_params": null,
            "state": "present",
            "tags": null,
            "token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "url": "https://demo.nautobot.com/",
            "validate_certs": true
        }
    },
    "msg": "cloud_resource_type Cisco Quantum Network Remix updated"
}
```

Reply:
Interesting that a lookup for the object would return it but not the module `.get`. We may need to investigate this. If this is just the "canary" of an underlying issue I'm fine with commenting it out (with a comment on why) for now and filing a bug issue and fixing it separately.

Reply:
Traced it back to pynautobot, here is a simple pynautobot script:

```python
crt = nb.cloud.cloud_resource_types.get(name="Cisco Quantum Network Remix")
print(crt.serialize())
```
Output:
```json
{"id": "1e3dba82-bbc0-4909-ae2d-840e88456d5c", "object_type": "cloud.cloudresourcetype", "display": "Cisco: Cisco Quantum Network Remix", "url": "http://192.168.10.200:8000/api/cloud/cloud-resource-types/1e3dba82-bbc0-4909-ae2d-840e88456d5c/", "natural_slug": "cisco-quantum-network-remix_1e3d", "content_types": ["cloud.cloudnetwork"], "name": "Cisco Quantum Network Remix", "description": "Very fast network reimagined!", "config_schema": "", "provider": "4130061e-f28e-4e46-b4b4-a71ffaa86555", "created": "2024-11-06T22:01:12.638908Z", "last_updated": "2024-11-06T22:15:00.331018Z", "notes_url": "http://192.168.10.200:8000/api/cloud/cloud-resource-types/1e3dba82-bbc0-4909-ae2d-840e88456d5c/notes/", "custom_fields": {}, "tags": []}
```

I'll see if I can figure out what's going on.  Right now, the core issue seems to be upstream.

Reply:
I have it fixed with some changes to pynautobot and confirmed idempotency works as expected.  Do you want to open a separate issue for these two items to fix it afterwards or work on getting a release dropped in pynautobot to fix it first?

Reply:
Let's get a pynautobot patch out to fix it